WcaOnRails/app/models/competition.rb

```diff
@@ -466,22 +466,76 @@ def country
   end
 
   def self.search(query, params: {})
-    sql_query = "%#{query}%"
-    Competition.where("id LIKE :sql_query OR name LIKE :sql_query OR cellName LIKE :sql_query OR cityName LIKE :sql_query OR countryId LIKE :sql_query", sql_query: sql_query).order(year: :desc, month: :desc, day: :desc)
+    competitions = Competition.where(showAtAll: true)
```

Comment:
Yeah, I get the attraction of an object that can both be a query and a result, but the price in confusion is probably bigger than the benefit in flexibility. Either way, that's the ActiveRecord we live with - though I hear there are better alternative Gems.

I wouldn't build a big string, but an array of clauses that you `.join(" AND ")` at the end. If AR wasn't so primitive, you could use the much cleaner Hash syntax, but this isn't MongoDB...

(I'm not asking you to change this, just chatting about what I would do.)